packages/edit-site/src/components/header/index.js

```diff
@@ -21,9 +22,21 @@ import TemplateSwitcher from '../template-switcher';
 import SaveButton from '../save-button';
 import UndoButton from './undo-redo/undo';
 import RedoButton from './undo-redo/redo';
+import FullscreenModeClose from './fullscreen-mode-close';
 
 const inserterToggleProps = { isPrimary: true };
 
+const CloseButton = () => {
+	const slot = useSlot( MainDashboardButton.slotName );
+	const hasFills = Boolean( slot.fills && slot.fills.length );
+
+	if ( ! hasFills ) {
+		return <FullscreenModeClose />;
+	}
+
+	return <MainDashboardButton.Slot bubblesVirtually />;
+};
```

Comment:
This should be defined in the slot's file.

Reply:
Moved in cbc8eba28bdb0ae229cbc7a54c0e4190eff813e9.